modules/Emails/EmailsController.php

```diff
@@ -139,19 +139,43 @@ public function action_ComposeView()
      */
     public function action_send()
     {
-        $this->bean = $this->bean->populateBeanFromRequest($this->bean, $_REQUEST);
-        $this->bean->save();
+        global $current_user;
+        global $app_strings;
 
-        $this->bean->handleMultipleFileAttachments();
+        $this->bean = $this->bean->populateBeanFromRequest($this->bean);
+        $inboundEmailAccount = new InboundEmail();
+        $inboundEmailAccount->retrieve($_REQUEST['inbound_email_id']);
 
-        if ($this->bean->send()) {
-            $this->bean->status = 'sent';
+        if($this->userIsAllowedToSendEmail($current_user, $inboundEmailAccount, $this->bean)) {
             $this->bean->save();
+
+            $this->bean->handleMultipleFileAttachments();
+
+            if ($this->bean->send()) {
+                $this->bean->status = 'sent';
+                $this->bean->save();
+            } else {
+                $this->bean->status = 'sent_error';
```

Comment:
phpdoc of $status says this is `dom_email_status` but that dropdown has `send_error` and not `sent_error`.
Will you take a look, or should we open a new issue to investigate this?

![screenshot from 2017-06-29 12 40 34](https://user-images.githubusercontent.com/329719/27685935-50def838-5cc8-11e7-9d8d-242e355f4155.png)

```bash
$ grep -Frni sent_error modules/*Email*
modules/Emails/views/view.sendemail.php:83:            case 'sent_error':
modules/Emails/EmailsController.php:146:            $this->bean->status = 'sent_error';

$ grep -Frni send_error modules/*Email*
modules/Emails/Save.php:189:            $focus->status = 'send_error';
modules/Emails/Email.php:781:            $this->status = 'send_error';
modules/Emails/Email.php:1153:                $this->status = 'send_error';
modules/Emails/Email.php:3076:            if ($this->type == 'out' && $this->status == 'send_error') {
modules/Emails/Email.php:3082:            if (($this->type == 'out' && $this->status == 'send_error') || $this->type == 'draft') {
```